src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/RedirectHandler.cs

```diff
@@ -0,0 +1,126 @@
+        }
+
+        protected internal override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            HttpResponseMessage response = await _initialInnerHandler.SendAsync(request, cancellationToken).ConfigureAwait(false);
+
+            uint redirectCount = 0;
+            Uri redirectUri;
+            while ((redirectUri = GetUriForRedirect(request.RequestUri, response)) != null)
+            {
+                response.Dispose();
+
+                redirectCount++;
+                if (redirectCount > _maxAutomaticRedirections)
+                {
+                    throw new HttpRequestException(SR.net_http_max_redirects);
+                }
+
+                // Clear the authorization header.
+                request.Headers.Authorization = null;
+
+                // Set up for the redirect
+                request.RequestUri = redirectUri;
+                if (RequestRequiresForceGet(response.StatusCode, request.Method))
+                {
+                    request.Method = HttpMethod.Get;
+                    request.Content = null;
+                }
+
+                // Issue the redirected request.
+                response = await _redirectInnerHandler.SendAsync(request, cancellationToken);
+            }
+
+            return response;
+        }
+
+        private Uri GetUriForRedirect(Uri requestUri, HttpResponseMessage response)
+        {
+            switch (response.StatusCode)
+            {
+                case HttpStatusCode.Moved:
+                case HttpStatusCode.Found:
+                case HttpStatusCode.SeeOther:
+                case HttpStatusCode.TemporaryRedirect:
+                case HttpStatusCode.MultipleChoices:
+                    break;
+
+                default:
+                    return null;
+            }
+
+            Uri location = response.Headers.Location;
+            if (location == null)
+            {
+                return null;
+            }
+
+            if (!location.IsAbsoluteUri)
+            {
+                location = new Uri(requestUri, location);
+            }
+
+            // Disallow automatic redirection from secure to non-secure schemes
+            if (HttpUtilities.IsSupportedSecureScheme(requestUri.Scheme) && !HttpUtilities.IsSupportedSecureScheme(location.Scheme))
+            {
+                return null;
+            }
+
+            return location;
+        }
+
+        private static bool RequestRequiresForceGet(HttpStatusCode statusCode, HttpMethod requestMethod)
+        {
+            if (statusCode == HttpStatusCode.Moved ||
+                statusCode == HttpStatusCode.Found ||
+                statusCode == HttpStatusCode.SeeOther ||
+                statusCode == HttpStatusCode.MultipleChoices)
+            {
+                return requestMethod == HttpMethod.Post;
+            }
+
+            return false;
+        }
```

Comment:
Nit: you'll get slightly better IL for this if you instead use a switch, as the compiler is then able to generate a simple jump table for the 300-303 values.
```C#
private static bool RequestRequiresForceGet(HttpStatusCode statusCode, HttpMethod requestMethod)
{
    switch (statusCode)
    {
        case HttpStatusCode.Moved:
        case HttpStatusCode.Found:
        case HttpStatusCode.SeeOther:
        case HttpStatusCode.MultipleChoices:
            return requestMethod == HttpMethod.Post;
        default:
            return false;
    }
}
```

Reply:
Ok